### DISCUSSION TITLE
Extracting contrasts for a multilevel factor of interest

### DISCUSSION THREAD

**Discussion by eoinoh91** [FrederickHuangLin/ANCOMBC#99] [Q&A]
Hello

I have a dataset with 2 factors of importance 
1. Treatment, which has 4 levels
2. TimePeriod

I want to test for differences between each of the 4 treatment groups (e.g. 1vs2, 1vs.3, 1vs.4, 2vs.3 etc etc.), while controlling for the effect of TimePeriod

How can I set this up correctly to extract each of the Treatment contrasts?

This is my code chunk for ancombc:

out = ancombc(phyloseq = ps.ancom.genus, formula = "Treatment + TimePeriod", zero_cut = 0.9,
              p_adj_method = "holm", lib_cut = 1000, 
              group = "Treatment", struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

Thanks!

**Comment by FrederickHuangLin** [comment]
Hi @eoinoh91,

The new `ancombc2` function can perform the pairwise directional test as you specified.

I just pushed the changes to the Bioconductor branches. It might take a few business days for the package to become available (but the `devel` version is ready here!).

Best,
Huang
